torch/utils/_ordered_set.py

```diff
@@ -174,3 +174,7 @@ def __setstate__(self, state: list[T]) -> None:
 
     def __reduce__(self) -> tuple[type[OrderedSet[T]], tuple[list[T]]]:
         return (OrderedSet, (list(self),))
+
+class ErrorOnUpdateOrderedSet(OrderedSet):
+    def update(self, *others):
+        raise NotImplementedError("Update operation is not allowed for this OrderedSet subclass.")
```

Comment:
The way this is structured is weird and will have unexpected IsInstance behavior...